pytorch_lightning/tuner/batch_size_scaling.py

```diff
@@ -217,6 +217,7 @@ def _run_binsearch_scaling(
                 high = new_size
                 midval = (high + low) // 2
                 new_size, _ = _adjust_batch_size(trainer, batch_arg_name, value=midval, desc="failed")
+                trainer.reset_train_dataloader(model)
```

Comment:
we usually also resets the validation dataloader, like here:
https://github.com/PyTorchLightning/pytorch-lightning/blob/ad32132c204ac9ee6003586e6ee44e2befa6db79/pytorch_lightning/tuner/batch_size_scaling.py#L168-L169

Reply:
As far as I can tell unless the very first batchsize fails the power scaling will have changed=TRUE from the previous size so even when it fails later on 'changed' is still TRUE so it'll always reset the train_dataloader.
Is the val_dataloader used in the scaling? I always assumed it was just the train_dataloader than was checked for the batch size. But yes, it's good to reset both.

Reply:
There should be a method to reset all dataloaders?